content/running_spark_apps_with_emr_on_spot_instances/selecting_instance_types.md

```diff
@@ -3,40 +3,65 @@ title: "Selecting instance types"
-Instead, we'll focus on instance types with lower interruption rates and suitable vCPU/Memory ratio. As an example, at the time of writing, in the EU (Ireland) region, these could be: r4.xlarge, r4.2xlarge, i3.xlarge, i3.2xlarge, r5d.xlarge
+{{% notice info %}}
+There are over 275 different instance types available on EC2 which can make the process of selecting appropriate instance types difficult. **[amazon-ec2-instance-selector](https://github.com/aws/amazon-ec2-instance-selector)** helps you select compatible instance types for your application to run on. The command line interface can be passed resource criteria like vCPUs, memory, network performance, and much more and then return the available, matching instance types.
+{{% /notice %}}
 
-![Spot Instance Advisor](/images/running-emr-spark-apps-on-spot/spotinstanceadvisor1.png)
+We will use **amazon-ec2-instance-selector** to help us select the relevant instance
+types with sufficient number of vCPUs and RAM.
 
-{{% notice note %}}
-Spot Instance interruption rates are dynamic, the above just provides a real world example from a specific time and would probably be different when you are performing this workshop.
-{{% /notice %}}
+Let's first install amazon-ec2-instance-selector on Cloud9 IDE:
+
+```
+curl -Lo ec2-instance-selector https://github.com/aws/amazon-ec2-instance-selector/releases/download/v1.3.0/ec2-instance-selector-`uname | tr '[:upper:]' '[:lower:]'`-amd64 && chmod +x ec2-instance-selector
+sudo mv ec2-instance-selector /usr/local/bin/
+ec2-instance-selector --version
+```
 
-To keep our flexibility in place and be able to provide multiple instance types for our EMR cluster, we need to make sure that our executor size will be under the EMR YARN limitation that we saw in the previous step, 
+Now that you have ec2-instance-selector installed, you can run
+`ec2-instance-selector --help`, to understand how you could use it for selecting
+instances that match your workload requirements.
 
-**Your first task**: Find and take note of 5 instance types in the region where you have created your VPC to run your EMR cluster, which will allow running executors with at least 4 vCPUs and 30+ GB of RAM, and also have low Spot interruption rates (maximum 10-15%).
+For the purpose of this workshop we will select instances based on below criteria -\
+ * Instances that have minimum 4 vCPUs and maximum 16 vCPUs\
+ * Instances which have vCPU to Memory ratio of 1:8, same as R Instance family\
+ * Instances with CPU Architecture x86_64 and no GPU Instances\
+ * Instances that belong to current generation\ 
+ * Instances types that are not supported by EMR such as R5N, R5ad and R5b. Enhanced z, I and D Instance families, which are priced higher than R family. So basically, adding a deny list with the regular expression `.*n.*|.*ad.*|.*b.*|^[zid].*`.

```

Comment:
Again, the bullet list doesn't come as well formatted. Please read the section regarding list with bullet points : https://guides.github.com/features/mastering-markdown/ and amend accordingly. `\` are causing issues with the carriage returns in some cases

![image](https://user-images.githubusercontent.com/46894578/103879649-0c9d7f00-50d0-11eb-8fd3-c768bf753716.png)


In this case this section should be changed to : 


```
For the purpose of this workshop we will select instances based on below criteria

 * Instances that have minimum 4 vCPUs and maximum 16 vCPUs
 * Instances which have vCPU to Memory ratio of 1:8, same as R Instance family
 * Instances with CPU Architecture x86_64 and no GPU Instances
 * Instances that belong to current generation
 * Instances types that are not supported by EMR such as R5N, R5ad and R5b. Enhanced z, I and D Instance families, which are priced higher than R family. So basically, adding a deny list with the regular expression `.*n.*|.*ad.*|.*b.*|^[zid].*`.
```